lib/AST/ASTImporter.cpp

```diff
@@ -1028,18 +1028,33 @@ void ASTNodeImporter::ImportDeclContext(DeclContext *FromDC, bool ForceImport,
     Importer.ImportContext(FromDC);
     return;
   }
-  llvm::SmallVector<Decl*, 8> Decls;
-  for (auto *From : FromDC->decls())
-    Decls.push_back(Importer.Import(From));
-  if (ToDC) {
-    // Restore the order.
-    for (auto *D : Decls) {
-      if (D && D->getLexicalDeclContext() == ToDC)
```

Comment:
Why did you remove the `D->getLexicalDeclContext() == ToDC` check?

Reply:
Does `containsDecl` checking the LexicalDeclContext or the semantic one?

Reply:
`containsDecl` do check for the lexical decl context. 
If the declcontext has only one decl than it checks against that, otherwise it just checks whether we have a list of decls but does search through the list. So I think this check includes your check.
```
bool DeclContext::containsDecl(Decl *D) const {
  return (D->getLexicalDeclContext() == this &&
          (D->NextInContextAndBits.getPointer() || D == LastDecl));
}
```